test/integration/scheduler_perf/scheduler_perf.go

```diff
@@ -684,28 +689,19 @@ func RunBenchmarkPerfScheduling(b *testing.B, outOfTreePluginRegistry frameworkr
 					// quit *before* restoring klog settings.
 					framework.GoleakCheck(b)
 
-					ctx := context.Background()
-
-					if *useTestingLog {
-						// In addition to redirection klog
-						// output, also enable contextual
-						// logging.
-						_, ctx = ktesting.NewTestContext(b)
-					}
-
 					// Now that we are ready to run, start
 					// etcd.
 					framework.StartEtcd(b, output)
 
 					// 30 minutes should be plenty enough even for the 5000-node tests.
-					ctx, cancel := context.WithTimeout(ctx, 30*time.Minute)
-					b.Cleanup(cancel)
+					timeout := 30 * time.Minute
+					tCtx = ktesting.WithTimeout(tCtx, timeout, fmt.Sprintf("timed out after the %s per-test timeout", timeout))
```

Comment:
```suggestion
					tCtx = ktesting.WithTimeout(tCtx, timeout, fmt.Sprintf("timed out after the %s per-test", timeout))
```

Reply:
This would lead to a message saying "timed out after the 30s per-test".

I think we should keep "per-test timeout" in the message.